calculator_test.go

```diff
@@ -173,19 +356,194 @@ func TestDivide(t *testing.T) {
+			want:        1,
+			nums:        []float64{20, 5, 2, 2},
+			errExpected: false,
+		},
+		{
+			name:        "More than 2 fractions",
+			want:        2,
+			nums:        []float64{5.5, 1.1, 2.5},
+			errExpected: false,
+		},
+		{
+			name:        "More than 2 fractions starting with 0",
+			want:        0,
+			nums:        []float64{0, 1.1, 2.5},
+			errExpected: false,
+		},
+		{
+			name:        "More than 2 fractions",
+			want:        0,
+			nums:        []float64{5.5, 0, 2.2},
+			errExpected: true,
+		},
+	}
+
+	for _, tc := range tcs {
+		got, err := calculator.DivideN(tc.nums)
+		errReceived := err != nil
+		if tc.errExpected != errReceived {
+			t.Errorf("Test DivideN (%v) (%s) failed: Unexpected error received",
+				tc.nums, tc.name)
+		} else {
+			if !tc.errExpected && !almostEqual(tc.want, got) {
+				t.Errorf("Test DivideN (%v) (%s) failed: want %f, got %f",
+					tc.nums, tc.name, tc.want, got)
+			}
+		}
+	}
+}
+
+func TestSquareRoot(t *testing.T) {
+	t.Parallel()
+	var tcs = []struct {
+		name        string
+		want        float64
+		a           float64
+		errExpected bool
+	}{
+		{
+			name:        "Squareroot of positive number",
+			want:        4,
+			a:           16,
+			errExpected: false,
+		},
+		{
+			name:        "Squareroot of 0",
+			want:        0,
+			a:           0,
+			errExpected: false,
+		},
+		{
+			name:        "Squareroot of negative number",
+			want:        0,
+			a:           -1,
+			errExpected: true,
+		},
+	}
+
+	for _, tc := range tcs {
+		got, err := calculator.SquareRoot(tc.a)
 		errReceived := err != nil
-		if test.errExpected != errReceived{
-				t.Errorf("Test Divide (%f, %f) %s failed: Unexpected error received",
-					test.a, test.b, test.name)
+		if tc.errExpected != errReceived {
+			t.Errorf("Test Squareroot (%f) (%s) failed: Unexpected error received",
+				tc.a, tc.name)
 		} else {
-			if !test.errExpected && !almostEqual(test.want, got) {
-				t.Errorf("Test Divide (%f, %f) %s failed: want %f, got %f",
-					test.a, test.b, test.name, test.want, got)
+			if !tc.errExpected && !almostEqual(tc.want, got) {
+				t.Errorf("Test Squareroot (%f) (%s) failed: want %f, got %f",
+					tc.a, tc.name, tc.want, got)
 			}
 		}
+	}
+}
 
+func TestComputeString(t *testing.T) {
+	t.Parallel()
+	var tcs = []struct {
+		name        string
+		input       string
+		want        float64
+		errExpected bool
+	}{
+		{
+			name:        "Addition of two numbers with no spaces",
+			input:       "2*2",
+			want:        4,
+			errExpected: false,
+		},
+		{
+			name:        "Subtraction with leading space",
+			input:       "10     -     2",
+			want:        8,
+			errExpected: false,
+		},
+		{
+			name:        "Multiplication with fraction",
+			input:       "    4 *1.5",
+			want:        6,
+			errExpected: false,
+		},
+		{
+			name:        "Division of two fraction",
+			input:       "    4.4 /2.2",
+			want:        2.0,
+			errExpected: false,
+		},
+		{
+			name:        "No operator",
```

Comment:
```suggestion
			name:        "Unsupported operator",
```

Reply:
Let's add a 'no operator' test case too! Something like `"2 2"`. (And see what other invalid inputs you can come up with!)